File: dailyproblem/06_03_2025/05_03_2025.py
class Solution:
    def findMissingAndRepeatedValues(self, grid):
        n = len(grid)
        num_set = set()
        repeated = -1
        total_sum = 0
        expected_sum = (n * n * (n * n + 1)) // 2  # Sum of numbers 1 to n^2

        for i in range(n):
            for j in range(n):
                val = grid[i][j]
                total_sum += val
                if val in num_set:
                    repeated = val  # Found the repeated number
                num_set.add(val)

        missing = expected_sum - (total_sum - repeated)  # Correct the sum by subtracting duplicate
        return [repeated, missing]

File: dailyproblem/06_03_2025/05_03_2025.cpp
class Solution {
    public:
        vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
            int n = grid.size();
            unordered_set<int> seen;
            int repeated = -1;
            long long totalSum = 0;
            long long expectedSum = (long long)(n * n) * (n * n + 1) / 2; // Sum of 1 to n^2
    
            for (int i = 0; i < n; ++i) {
                for (int j = 0; j < n; ++j) {
                    int val = grid[i][j];
                    totalSum += val;
                    if (seen.count(val)) {
                        repeated = val;  // Found duplicate
                    }
                    seen.insert(val);
                }
            }
    
            int missing = expectedSum - (totalSum - repeated);
            return {repeated, missing};
        }
    };
    